src/java/org/apache/cassandra/auth/Roles.java

```diff
@@ -189,4 +189,11 @@ public static Role fromRoleResource(RoleResource resource, IRoleManager roleMana
                                   .map(RoleResource::getRoleName)
                                   .collect(Collectors.toSet()));
    }
+
+    public static Set<RoleResource> getAllSuperUsers()
+    {
+        return getAllRoles().stream()
```

Comment:
maybe it would be cool to have a method `getAllRoles(Predicate<RoleResource> p)` in Roles.java like this:

    public static Set<RoleResource> getAllRoles(Predicate<RoleResource> p)
    {
        return getAllRoles().stream().filter(p).collect(Collectors.toUnmodifiableSet());
    }

That way we would not have a very specific method about `getAllSuperUsers` but developers would have more freedom to filter roles however they please ... e.g. "give me all roles with so and so parents" or applicable permissions or role names etc ...

Reply:
Done